prefect_dask/task_runners.py

```diff
@@ -209,11 +209,17 @@ async def submit(
         # where possible to optimize Dask task scheduling
         call_kwargs = self._optimize_futures(call.keywords)
 
+        if "task_run" in call_kwargs:
+            task_run = call_kwargs["task_run"]
+            dask_key = f"{task_run.name}-{task_run.id.hex}"
```

Comment:
Because we mark this function as `pure` this key is used to prevent duplicate runs. However, these items will match if the task run failed and is running again for a retried flow run resulting in a retrieval from the Dask cache. You'll need to make sure the key includes the flow run count as it did before.